galasa-extensions-parent/dev.galasa.ras.couchdb/src/test/java/dev/galasa/ras/couchdb/internal/mocks/MockCloseableHttpClient.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package dev.galasa.ras.couchdb.internal.mocks;
+
+import static org.assertj.core.api.Fail.fail;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+
+import org.apache.http.HttpHost;
+import org.apache.http.HttpRequest;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.conn.*;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.params.*;
+import org.apache.http.protocol.HttpContext;
+
+public class MockCloseableHttpClient extends CloseableHttpClient {
+
+    private Iterator<HttpInteraction> interactionWalker;
+    private HttpInteraction currentInteraction ;
+
+    public MockCloseableHttpClient( List<HttpInteraction> interactions) {
+        this.interactionWalker = interactions.iterator();
+        nextInteraction();
+    }
+
+    // Bump over to the next interaction.
+    private void nextInteraction() {
+        if (interactionWalker.hasNext()) {
+            this.currentInteraction = interactionWalker.next();
+        } else {
+             this.currentInteraction = null ;
+        }
+    }
+
+    @Override
+    @SuppressWarnings("deprecation")
+    public HttpParams getParams() {
+        // TODO Auto-generated method stub
```

Comment:
Very minor but I'd remove these TODOs - same applies in other mocks in this PR

Reply:
all fixed